Tests/ArcGISToolkitTests/PreplannedMapModelTests.swift

```diff
@@ -284,6 +284,81 @@ class PreplannedMapModelTests: XCTestCase {
         
         model2.status.assertExpectedValue(.downloaded)
     }
+    
+    @MainActor
+    func testLoadMobileMapPackage() async throws {
+        let portalItem = PortalItem(portal: Portal.arcGISOnline(connection: .anonymous), id: .init("acc027394bc84c2fb04d1ed317aac674")!)
+        let task = OfflineMapTask(portalItem: portalItem)
+        let areas = try await task.preplannedMapAreas
+        let area = try XCTUnwrap(areas.first)
+        let areaID = try XCTUnwrap(area.portalItem.id)
+        
+        defer {
+            // Clean up JobManager.
+            OfflineManager.shared.jobManager.jobs.removeAll()
+            
+            // Clean up folder.
+            let directory = FileManager.default.preplannedDirectory(
+                forPortalItemID: portalItem.id!,
+                preplannedMapAreaID: areaID
+            )
+            try? FileManager.default.removeItem(at: directory)
+        }
+        
+        let model = PreplannedMapModel(
+            offlineMapTask: task,
+            mapArea: area,
+            portalItemID: portalItem.id!,
+            preplannedMapAreaID: areaID,
+            // User notifications in unit tests are not supported, must pass false here
+            // or the test process will crash.
+            showsUserNotificationOnCompletion: false
+        )
+        
+        var statuses = [PreplannedMapModel.Status]()
+        var subscriptions = Set<AnyCancellable>()
+        model.$status
+            .receive(on: DispatchQueue.main)
+            .sink { value in
+                statuses.append(value)
+            }
+            .store(in: &subscriptions)
+        
+        await model.load()
+        
+        // Start downloading
+        await model.downloadPreplannedMapArea()
+        
+        // Wait for job to finish.
+        _ = await model.job?.result
+        
+        // Verify that mobile map package can be loaded.
+        let map = await model.loadMobileMapPackage()
+        XCTAssertNotNil(map)
+        
+        // Give the final status some time to be updated.
+        try? await Task.sleep(nanoseconds: 1_000_000)
+        
+        // Verify statuses
+        let expectedStatusCount = 6
+        guard statuses.count == expectedStatusCount else {
+            XCTFail("Expected a statuses count of \(expectedStatusCount), count is \(statuses.count).")
+            return
+        }
+        
+        let expected: [PreplannedMapModel.Status] = [
+            .notLoaded,
+            .loading,
+            .packaged,
+            .downloading,
+            .downloading,
+            .downloaded
+        ]
+        
+        for (status, expected) in zip(statuses, expected) {
+            status.assertExpectedValue(expected)
+        }
```

Comment:
`PreplannedMapModel.Status` not being equatable makes it _very_ hard to work with. That's one reason why we didn't do something similar with `LoadStatus`. If at all possible, I strongly suggest moving the associated values out of the enum, allowing it to be equatable and greatly simplifying any code that uses it.

Reply:
I added equatable conformance to `PreplannedMapModel.Status` and kept the associated values on the enum because it simplifies testing. The helper methods for `PreplannedMapModel.Status` in `PreplannedMapModelTests` have been removed.